t/50reverse-proxy-informational.t

```diff
@@ -0,0 +1,110 @@
+use strict;
+use warnings;
+use Net::EmptyPort qw(check_port empty_port);
+use Test::More;
+use URI::Escape;
+use t::Util;
+
+plan skip_all => 'curl not found'
+    unless prog_exists('curl');
+plan skip_all => 'plackup not found'
+    unless prog_exists('plackup');
+plan skip_all => 'Starlet not found'
+    unless system('perl -MStarlet /dev/null > /dev/null 2>&1') == 0;
+
+my $upstream_port = empty_port();
+my $upstream = spawn_server(
+    argv     => [ qw(plackup -s Starlet --access-log /dev/null --listen), $upstream_port, ASSETS_DIR . "/upstream.psgi" ],
+    is_ready =>  sub {
+        check_port($upstream_port);
+    },
+);
+
+subtest '100 Continue' => sub {
+    doit(100);
+};
+subtest '103 Early Hints' => sub {
+    doit(103);
+};
+done_testing();
+
+sub doit {
+    my ($status) = @_;
+    my $link = $status == 103 ? 1 : 0;
+
+    subtest 'all' => sub {
+        my $server = spawn_h2o(<< "EOT");
+send-informational: all
+hosts:
+  default:
+    paths:
+      /:
+        proxy.reverse.url: http://127.0.0.1:$upstream_port
+      /tweak-headers:
+        header.unset:
+          header: "link"
+          when: early
+        header.add:
+          header:
+            - "foo: FOO"
+            - "bar: BAR"
+          when: early
+        proxy.reverse.url: http://127.0.0.1:$upstream_port
+EOT
+        run_with_curl($server, sub {
+            my ($proto, $port, $curl) = @_;
+            my $resp;
+            $resp = `$curl --silent --dump-header /dev/stdout '$proto://127.0.0.1:$port/1xx?status=$status&link=$link'`;
+            like $resp, qr{^HTTP/[\d.]+ $status}mi;
+            (my $eh, $resp) = split(/\r\n\r\n/, $resp, 2);
+            like $eh, qr{^link: </index.js>; rel=preload}mi if $link;
+    
+            if ($status == 103) {
```

Comment:
Don't we need to test that the server is actually sending 103 here?

I believe that the test would succeed even if the server did not send 103, all the 103-related tests are under `if ($status == 103)`.

PS. In particular, I am thinking how we test the behavior of the HTTP/2 protocol implementation.